src/Core/Components/Switch/FluentSwitch.razor

```diff
@@ -0,0 +1,18 @@
+﻿@namespace Microsoft.FluentUI.AspNetCore.Components
+@using Microsoft.FluentUI.AspNetCore.Components.Extensions
+@inherits FluentInputBase<bool>
+
+<FluentField InputComponent="@this" ForId="@Id" Class="@ClassValue" Style="@((ReadOnly ? "pointer-events: none;" : "") + StyleValue)">
```

Comment:
You can't hard-code your style: you have to use the `DefaultStyleBuilder` as we have done in other components (ex. FluentButton).
But in this case, this style is pre-defined" ans always the same: you need to add a  FluentSwitch.razor.css` containing `fluent-switch[readonly] { }`.

Reply:
Style removed from field.